pandas/io/excel/_xlrd.py

```diff
@@ -98,9 +106,16 @@ def _parse_cell(cell_contents, cell_typ):
                     cell_contents = val
             return cell_contents
 
-        data = []
+        data: List[List[Scalar]] = []
 
-        for i in range(sheet.nrows):
+        sheet_nrows = sheet.nrows
+        if nrows is not None and isinstance(header, int) and isinstance(skiprows, int):
+            sheet_nrows = min(header + skiprows + nrows + 1, sheet_nrows)
+
+        for i in range(sheet_nrows):
+            if self.should_skip_row(i, header, skiprows, nrows):
+                data.append([])
```

Comment:
Do we need to append the empty list here? Would be preferable to just continue

Reply:
I removed `should_skip_row` entirely, I think the benefit from the original PR came from not reading all the rows into memory, rather than from optimising skipping rows